lib/webdriverio.js

```diff
@@ -208,14 +208,33 @@ let WebdriverIO = function (args, modifier) {
         return commands && commands[commands.length - 1].name === command
     }
 
+    function setConnectionRetryOpts (client, opts) {
+        if (typeof opts !== 'object') {
+            return
+        }
+
+        const {connectionRetryTimeout: defaultTimeout, connectionRetryCount: defaultCount} = options
+        const {connectionRetryTimeout = defaultTimeout, connectionRetryCount = defaultCount} = opts
+
+        const connectionOpts = {connectionRetryTimeout, connectionRetryCount}
+
+        client.requestHandler.updateConnectionOpts(connectionOpts)
+    }
+
     function takeScreenshot (err) {
         const client = unit()
         const failDate = new Date()
 
         // don't take a new screenshot if we already got one from Selenium
         const getScreenshot = typeof err.screenshot === 'string'
             ? () => err.screenshot
-            : () => rawCommands.screenshot.call(client).then((res) => res.value)
+            : () => {
+                setConnectionRetryOpts(client, options.screenshotOnReject)
```

Comment:
Lets return previous connection retry options values from `setConnectionRetryOpts`, store them inside some variable and then set them back. Like
```js
const prevRetryOpts = setConnectionRetryOpts(client, options.screenshotOnReject)
...
    .finally(() => setConnectionRetryOpts(client, prevRetryOpts))
```